compiler/noirc_frontend/src/ast/mod.rs

```diff
@@ -112,6 +112,9 @@ pub enum UnresolvedTypeData {
         /*env:*/ Box<UnresolvedType>,
     ),
 
+    // The type of quoted code for metaprogramming
+    Code,
```

Comment:
We had a `Type::Code` but not a `UnresolvedTypeData::Code` before so I needed to have this to have something to convert `Type::Code` to.